includes/REST_API/Media_Hotlinking_Controller.php

```diff
@@ -0,0 +1,290 @@
+
+		$data = [];
+
+		$check_fields = array_keys( $link );
+		foreach ( $check_fields as $check_field ) {
+			if ( rest_is_field_included( $check_field, $fields ) ) {
+				$data[ $check_field ] = rest_sanitize_value_from_schema( $link[ $check_field ], $schema['properties'][ $check_field ] );
+			}
+		}
+
+		$context = ! empty( $request['context'] ) ? $request['context'] : 'view';
+		$data    = $this->add_additional_fields_to_object( $data, $request );
+		$data    = $this->filter_response_by_context( $data, $context );
+
+		// Wrap the data in a response object.
+		$response = rest_ensure_response( $data );
+
+		return $response;
+	}
+
+	/**
+	 * Retrieves the link's schema, conforming to JSON Schema.
+	 *
+	 * @since 1.11.0
+	 *
+	 * @return array Item schema data.
+	 */
+	public function get_item_schema(): array {
+		if ( $this->schema ) {
+			return $this->add_additional_fields_schema( $this->schema );
+		}
+
+		$schema = [
+			'$schema'    => 'http://json-schema.org/draft-04/schema#',
+			'title'      => 'link',
+			'type'       => 'object',
+			'properties' => [
+				'ext'       => [
+					'description' => __( 'File extension', 'web-stories' ),
+					'type'        => 'string',
+					'context'     => [ 'view', 'edit', 'embed' ],
+				],
+				'file_name' => [
+					'description' => __( 'File name', 'web-stories' ),
+					'type'        => 'string',
+					'context'     => [ 'view', 'edit', 'embed' ],
+				],
+				'file_size' => [
+					'description' => __( 'File size', 'web-stories' ),
+					'type'        => 'integer',
+					'context'     => [ 'view', 'edit', 'embed' ],
+				],
+				'mime_type' => [
+					'description' => __( 'Mime Type', 'web-stories' ),
+					'type'        => 'string',
+					'context'     => [ 'view', 'edit', 'embed' ],
+				],
+			],
+		];
+
+		$this->schema = $schema;
+
+		return $this->add_additional_fields_schema( $this->schema );
+	}
+
+	/**
+	 * Checks if current user can process links.
+	 *
+	 * @since 1.11.0
+	 *
+	 * @return true|WP_Error True if the request has read access, WP_Error object otherwise.
+	 */
+	public function parse_url_permissions_check() {
+		if ( ! $this->get_post_type_cap( Story_Post_Type::POST_TYPE_SLUG, 'edit_posts' ) ) {
+			return new WP_Error( 'rest_forbidden', __( 'Sorry, you are not allowed to insert external media.', 'web-stories' ), [ 'status' => rest_authorization_required_code() ] );
+		}
+
+		return true;
+	}
+
+	/**
+	 * Callback to validate urls.
+	 *
+	 * @since 1.11.0
+	 *
+	 * @param mixed $value Value to be validated.
+	 *
+	 * @return true|WP_Error
+	 */
+	public function validate_url( $value ) {
+		$url = untrailingslashit( $value );
+
+		if ( empty( $url ) ) {
+			return new WP_Error( 'rest_invalid_url', __( 'Invalid URL', 'web-stories' ), [ 'status' => 400 ] );
+		}
+
+		$parts = wp_parse_url( $url );
+
+		if ( ! isset( $parts['host'] ) ) {
+			return new WP_Error( 'rest_invalid_url_host', __( 'Invalid URL Host', 'web-stories' ), [ 'status' => 400 ] );
+		}
```

Comment:
Is there ever a case where this happens? 🤔  The REST API bails very early on invalid URLs already, so I don't see how/when this would be triggered.

Reply:
> The REST API bails very early on invalid URLs already, so I don't see how/when this would be triggered.

See. https://github.com/google/web-stories-wp/pull/8731#discussion_r692816954

Relative paths, are a valid url. In my testing, it doesn't trigger an error when I passed, as `/image/test.png`. 

It will become clear, once I have unit test.